cmake/modules/RootMacros.cmake

```diff
@@ -587,6 +587,11 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
   #---Get the library and module dependencies-----------------
   if(ARG_DEPENDENCIES)
     foreach(dep ${ARG_DEPENDENCIES})
+      if(NOT TARGET G__${dep})
+        # This is a library that doesn't come with dictionary/pcm
+        continue()
```

Comment:
Not all dictionary targets are in `G__*` form. When using the "new" method of adding a source to the library with `ROOT_GENERATE_DICTIONARY`, it generates a source file and adds to the normal library target instead of creating an explicit  target for the dictionary. However, you still have a dictionary/pcm for that library.